docs/misc-operations/analyze-update-verify-logs.md

```diff
@@ -0,0 +1,52 @@
+# When?
+
+When update verify tasks fail it is your responsibility as releaseduty to analyze them and determine whether or not any action needs to be taken for any differences found.
+
+# How? 
+
+Update verify tasks that have failed usually have a `diff-summary.log` in their artifacts. This file shows you all of the differences found for each update tested. In the diffs, `source` is an older version Firefox that a MAR file from the current release has been applied to, and `target` is the full installer for the current release.
+
+Here's an example of a very alarming difference:
+```
+Found diffs for complete update from https://aus5.mozilla.org/update/3/Firefox/59.0/20180215111455/WINNT_x86-msvc/en-US/beta-localtest/default/default/default/update.xml?force=1
+
+Files source/bin/xul.dll and target/bin/xul.dll differ
+```
+
+In the above log, `xul.dll` is shown to be different between an applied MAR and a full installer. If we were to ship a release with a difference like this, partial MARs would fail to apply for many users in the _next_ release. Usually a case like this represents an issue in the build system or release automation, and requires a rebuild. If you're not sure how to proceed, ask for help.
+
+If no `diff-summary.log` is attached to the Task something more serious went wrong. You will need to have a look at live.log to investigate.
+
+# Known differences
+
+There are two known cases where diffs are expected, and ignorable.
+
+## Beta channel/ACCEPTED_MAR_CHANNEL_IDS during secondary update verify in RCs
+
+```
+Found diffs for complete update from https://aus5.mozilla.org/update/3/Firefox/59.0/20180215111455/Linux_x86-gcc3/en-US/beta-localtest/default/default/default/update.xml?force=1
+diff -r source/firefox/defaults/pref/channel-prefs.js target/firefox/defaults/pref/channel-prefs.js
+5c5
+< pref("app.update.channel", "beta");
+---
+> pref("app.update.channel", "release");
+diff -r source/firefox/update-settings.ini target/firefox/update-settings.ini
+5c5
+< ACCEPTED_MAR_CHANNEL_IDS=firefox-mozilla-beta,firefox-mozilla-release
+---
+> ACCEPTED_MAR_CHANNEL_IDS=firefox-mozilla-release
```

Comment:
Cool. I think we should be able to add a wrapper to ignore these diffs for secondary-update-verify tasks. Or, better yet, raise if they're *not* diffing this way.

Reply:
👍

Reply:
Yeah, this would be great. I don't think we should _always_ be succeeding on diff though, even for secondary update verify. @catlee and I talked about this a little bit, and thought that it would be a lot easier to do whitelisting or blacklisting if we had machine readable diffs. It's really tough to accurately parse plain `diff` output, although not impossible.